docs/developer/architecture/authentication.rst

```diff
@@ -108,3 +109,180 @@ Open questions
   - the fact that most consumers of Prometheus stats are not human
     (e.g. Grafana, a federating Prometheus, scripts and others), hence not
     well-suited for performing the OIDC flow
+
+Design Choices
+--------------
+
+`Dex <https://github.com/dexidp/dex/>`_ is chosen as an Identity Provider(IdP)
+in MetalK8s based on the above :ref:`authentication-requirements` enforced
+by the following reasons:
```

Comment:
_enforced by the following reasons_ doesn't make much sense to me. Maybe just use _for the following reasons_